File: leetcode_118/solution.py
class Solution:
    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        ret = [
            [1 for _ in range(k + 1)]
            for k in range(numRows)
        ]
        for i in range(numRows):
            for j in range(1, i):
                ret[i][j] = ret[i - 1][j - 1] + ret[i - 1][j]
        return ret


if __name__ == '__main__':
    row = 5
    print(Solution().generate(row))

File: leetcode_118/solution.go
package main

import "fmt"

func generate(numRows int) [][]int {
	tri := make([][]int, numRows)
	for i := 0; i < numRows; i++ {
		lenRow := i + 1
		tri[i] = make([]int, lenRow)
		tri[i][0], tri[i][lenRow - 1] = 1, 1
		for j := 1; j < lenRow - 1; j++ {
			tri[i][j] = tri[i-1][j] + tri[i - 1][j - 1]
		}
	}
	return tri
}

func main() {
	numRows := 5
	fmt.Println(generate(numRows))
}
